src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/AS_StrandOddsRatio.java

```diff
@@ -0,0 +1,70 @@
+ *
+ * $$ refRatio = \frac{max(X[0][0], X[0][1])}{min(X[0][0], X[0][1} $$
+ *
+ * <p>and </p>
+ *
+ * $$ altRatio = \frac{max(X[1][0], X[1][1])}{min(X[1][0], X[1][1]} $$
+ *
+ * <p>ensures that the annotation value is large only. </p>
+ *
+ * <p>See the <a href="http://www.broadinstitute.org/gatk/guide/article?id=4732">method document on statistical tests</a> for a more detailed explanation of this statistical test.</p>
+ *
+ * <h3>Caveat</h3>
+ * <p>
+ * The name AS_StrandOddsRatio is not entirely appropriate because the implementation was changed somewhere between the start of development and release of this annotation. Now SOR isn't really an odds ratio anymore. The goal was to separate certain cases of data without penalizing variants that occur at the ends of exons because they tend to only be covered by reads in one direction (depending on which end of the exon they're on), so if a variant has 10 ref reads in the + direction, 1 ref read in the - direction, 9 alt reads in the + direction and 2 alt reads in the - direction, it's actually not strand biased, but the FS score is pretty bad. The implementation that resulted derived in part from empirically testing some read count tables of various sizes with various ratios and deciding from there.</p>
+ *
+ * <h3>Related annotations</h3>
+ * <ul>
+ *     <li><b><a href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_annotator_StrandOddsRatio.php">StrandOddsRatio</a></b> outputs a version of this annotation that includes all alternate alleles in a single calculation.</li>
+ *     <li><b><a href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_annotator_StrandBiasBySample.php">StrandBiasBySample</a></b> outputs counts of read depth per allele for each strand orientation.</li>
+ *     <li><b><a href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_annotator_FisherStrand.php">FisherStrand</a></b> uses Fisher's Exact Test to evaluate strand bias.</li>
+ * </ul>
+ *
+ */
+public final class AS_StrandOddsRatio extends AS_StrandBiasTest implements AS_StandardAnnotation {
+
+    @Override
+    public List<String> getKeyNames() {
+        return Collections.singletonList(GATKVCFConstants.AS_STRAND_ODDS_RATIO_KEY);
+    }
+
+    @Override
+    protected Map<String, Object> calculateAnnotationFromLikelihoodMap(final Map<String, PerReadAlleleLikelihoodMap> stratifiedPerReadAlleleLikelihoodMap,
+                                                                       final VariantContext vc){
+        // either SNP with no alignment context, or indels: per-read likelihood map needed
+        final int[][] table = getContingencyTable(stratifiedPerReadAlleleLikelihoodMap, vc, MIN_COUNT);
+        final double ratio = StrandOddsRatio.calculateSOR(table);
+        return Collections.singletonMap(getKeyNames().get(0), StrandOddsRatio.formattedValue(ratio));
```

Comment:
Should there be an interface method for formatting values? Seems like several of the annotations have code like this.